storage/database/dynamodb.go

```diff
@@ -125,26 +129,34 @@ func NewDynamoDB(config *DynamoDBConfig) (*dynamoDB, error) {
 		config.Endpoint = "https://dynamodb." + config.Region + ".amazonaws.com"
 	}
 
-	logger.Info("creating s3FileDB ", "bucket", config.TableName)
+	dynamoOpenedDBNum++
+
 	s3FileDB, err := newS3FileDB(config.Region, "https://s3."+config.Region+".amazonaws.com", config.TableName)
 	if err != nil {
 		logger.Error("Unable to create/get S3FileDB", "DB", config.TableName)
 		return nil, err
 	}
 
-	dynamoDB := &dynamoDB{
-		config:  config,
-		logger:  logger.NewWith("region", config.Region, "tableName", config.TableName),
-		quitCh:  make(chan struct{}),
-		writeCh: make(chan *batchWriteWorkerInput, itemChanSize),
-		fdb:     s3FileDB,
-		db: dynamodb.New(session.Must(session.NewSessionWithOptions(session.Options{
+	dynamoOnceWorker.Do(func() {
+		dynamoDBClient = dynamodb.New(session.Must(session.NewSessionWithOptions(session.Options{
 			Config: aws.Config{
 				Endpoint: aws.String(config.Endpoint),
 				Region:   aws.String(config.Region),
 			},
-		}))),
+		})))
+		dynamoWriteCh = make(chan *batchWriteWorkerInput, itemChanSize)
+		for i := 0; i < WorkerNum; i++ {
+			go createBatchWriteWorker(dynamoWriteCh)
+		}
+		logger.Info("made dynamo batch write workers", "workerNum", WorkerNum)
```

Comment:
Why did you move this logic?

Reply:
I changed as it was previous placed. PTAL